File: 02_Hollow rectangle/hollow_rectangle.c
#include<stdio.h>

int main(int argc, char const *argv[])
{
    int rows, columns;

    printf("Enter Rows: ");
    scanf("%d", &rows);

    printf("Enter Columns: ");
    scanf("%d", &columns);

    for(int i=0; i<rows; i++){
        for(int j=0; j<columns; j++){
            if(i==0 || i==(rows-1) || j==0 || j==(columns-1)){
                printf("* ");
            }else{
                printf("  ");
            }
        }
        printf("\n");
    }

    return 0;
}

File: 02_Hollow rectangle/hollow_rectangle.py
rows = int(input("Enter rows: "))
columns = int(input("Enter columns: "))

# Method_1 for printing hollow rectangle.
for i in range(0, rows):
    for j in range(0, columns):
        if(i == 0 or i==(rows-1) or j==0 or j==(columns-1)):
            print("*", end=" ")
        else:
            print(" ", end=" ")
    print("\n", end="")


# Method_2 for printing hollow rectangle.
for i in range(0, rows):
    for j in range(0, columns):
        if(i == 0 or i==(rows-1)):
            print("*", end=" ")
        elif(i>0 and i<(rows-1)):
            print("*", end=" ")
            print("  " * (columns-2),end="")
            print("*",end=" ")
            break
    print("\n", end="")

    """
    OUTPUT:

    * * * * * *
    *         *
    *         *
    * * * * * *
    
    """